main/src/com/google/refine/operations/column/ColumnAdditionByFetchingURLsOperation.java

```diff
@@ -321,14 +324,15 @@ Serializable cachedFetch(String urlString) {
         }
 
         Serializable fetch(String urlString, Header[] headers) {
-            try { // HttpClients.createDefault()) {
-                try {
-                    return _httpClient.getAsString(urlString, headers);
-                } catch (IOException e) {
-                    return _onError == OnError.StoreError ? new EvalError(e) : null;
-                }
-            } catch (Exception e) {
-                return _onError == OnError.StoreError ? new EvalError(e.getMessage()) : null;
+            try {
+                // Do a little dance to get all the pieces of the URL correctly encoded, if possible
+                // TODO: Should we try to undo any existing percent encoding, so we don't end up double encoded?
```

Comment:
I suspect users might have got used to doing the percent encoding on their side given that OpenRefine wasn't doing it for them, so I think it'd be really valuable to avoid double encoding here.

To test this more easily, it might be convenient to extract the URL parsing logic into a method that can be called independently from the unit test, without having to fiddle with the mock test server (but it's great that you also adapted that test).

Reply:
I've made the requested change.